File: back/src/main/java/fr/autostopfrance/Autostop/models/Trip.java
package fr.autostopfrance.Autostop.models;

import javax.persistence.*;

import com.google.maps.model.LatLng;

import java.util.ArrayList;

@Entity
@Table(name="trip")
public class Trip {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tripId")
    private Long id;
    @Column(name="origin")
    private LatLng origin;
    @Column(name="destination_string")
    private String destinationString;
    @Column(name="destination_lat_lng")
    private LatLng destinationLatLng;
    @Column(name="accepted")
    private boolean accepted;
    @Column(name="itinerary", columnDefinition = "text")
    private String itinerary;
//    private int 

    public Trip () {
    }

    public Trip(LatLng origin, String destinationString, LatLng destinationLatLng, String itinerary){
        this.origin = origin;
        this.destinationString = destinationString;
        this.destinationLatLng = destinationLatLng;
    	this.itinerary = itinerary;
    }

    public LatLng getOrigin() {
        return origin;
    }

    public void setOrigin(LatLng origin) {
        this.origin = origin;
    }

    public String getDestinationString() {
        return destinationString;
    }

    public void setDestinationString(String destination) {
        this.destinationString = destination;
    }
    
    public LatLng getDestinationLatLng() {
        return destinationLatLng;
    }

    public void setDestinationLatLng(LatLng destinationLatLng) {
        this.destinationLatLng = destinationLatLng;
    }

    public boolean isAccepted() {
        return accepted;
    }

    public void setAccepted(boolean accepted) {
        this.accepted = accepted;
    }

    public String getItinerary() {
        return itinerary;
    }

    public void setItinerary(String itinerary) {
        this.itinerary = itinerary;
    }

    public ArrayList<Object> calculateTrip(){
        return null;
    }
}

File: Front/autostop/src/models/Trip.ts
import { LatLng, ILatLng } from "@ionic-native/google-maps";

export class Trip {
    origin: LatLng;
    destinationString: string;
    destinationLatLng: ILatLng;
    itinerary: LatLng[];
    accepted: boolean;

  constructor(
    origin?: LatLng,
    destinationString?: string,
    destinationLatLng?: ILatLng,
    itinerary?: LatLng[],
  )

  {
    this.origin = origin;
    this.destinationString = destinationString;
    this.destinationLatLng = destinationLatLng;
    this.itinerary = itinerary;
    this.accepted = false;
  }

  public getOrigin(): LatLng {
    return this.origin;
  }

  public setOrigin(origin: LatLng): void {
    this.origin = origin;
  }


  public getDestinationString(): string {
    return this.destinationString;
  }

  public setDestinationString(destinationString: string): void {
    this.destinationString = destinationString;
  }


  public getDestinationLatLng(): ILatLng {
    return this.destinationLatLng;
  }

  public setDestinationLatLng(destinationLatLng: ILatLng): void {
    this.destinationLatLng = destinationLatLng;
  }

  public getItinerary(): LatLng[] {
    return this.itinerary;
  }

  public setItinerary(itinerary: LatLng[]): void {
    this.itinerary = itinerary;
  }
  public isAccepted(): boolean {
    return this.accepted;
  }

  public setAccepted(accepted: boolean): void {
    this.accepted = accepted;
  }

}
